typeahead: Fix bug where typeahead showed momentarily on shift + tab.

Joelute opened the pull request:
Break on `keyup` for case of `shift` key, instead of letting it trigger the typeahead.

This is a PR continuing on #24181. It was previously reverted due to a regression and it has now been fixed with #24812.

Fixes: #24152.

**Note:** Currently in main, we are calling `lookup` whenever we are letting go of the shift key to help fix the position of the typeahead. With the shift key being disabled in the changes, this becomes a problem as the typeahead fails to fix its position and will hover higher than usual. Thus, we will call `lookup` whenever we partially complete with `>` instead.

<!-- Describe your pull request here.-->


<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

<details>
<summary>Before:</summary>

![f5a72a0c5fe2b137fc81dde61dd47179](https://user-images.githubusercontent.com/62449508/227443181-1e7d5adc-01b9-4a8d-926b-d7ccafec2023.gif)
</details>
<details>
<summary>After:</summary>

![7361d279d6046cc2147f70943e13ecfe](https://user-images.githubusercontent.com/62449508/227424192-16732b76-8649-41ec-a5d0-917188fe20d5.gif)
</details>
<details>
<summary>Compose box:</summary>

![60d9ecd7cd1d3f21e1acaabfd6209bc3](https://user-images.githubusercontent.com/62449508/227442637-d284bbb2-cd9d-4145-b1e2-1e714bfb4a27.gif)
![7fe00276ac8e3b53698aec244e7b3bdf](https://user-images.githubusercontent.com/62449508/227442843-8e941cdb-5395-4078-864d-bfe41fac8ed4.gif)

</details>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

alya commented:
With this PR, the topic typeahead doesn't show up right away with `#**stream name>`, and I think we want it to. (It currently does on CZO.)

Joelute commented:
Yup, I have a fix for it ready.

Joelute commented:
There seems to be another problem where the typeahead relies on the press of an second key to correct it's position. Since this PR disables the shift key, the typeahead isn't able to fix it's position and hovers above the box.

alya commented:
@Joelute Is this PR ready for another review, then? If so, I think @N-Shar-ma can take a look next.

Joelute commented:
Yup it is, @N-Shar-ma please have a look :)

N-Shar-ma reviewed (changes requested):
I looked deeper into why the position issue was cropping up since calling the expensive `lookup` function seemed overkill for just fixing that. Turns out it was due to the header element of the typeahead not being hidden before calculation of the position based on container height.

So instead of calling `lookup`, just move up the `header` hiding / showing code in `show` above the code calculating `top_pos` for positioning the typeahead. I've tested this locally!

Joelute commented:
Thank you for looking into that issue @N-Shar-ma! I've added it in a sperate commit and moved that section of the code to the top of the function so it flows better. This looks good to me!

N-Shar-ma commented:
Looks good to me now, just switch the order of the 2 commits (the one shifting the header code should be like a prep commit before the other)

@alya Over to you now!

alya commented:
Works for me in manual testing!

@timabbott This PR has been reviewed by @N-Shar-ma and me -- please take a look.

timabbott commented:
Merged, thanks for the work on this @Joelute and @N-Shar-ma!
